rfc/0003-policy-signing.md

```diff
@@ -0,0 +1,440 @@
+
+Our policies are binary blobs, that makes hard to ensure the policy we're about to run
+matches with the source code tagged inside of a git repository.
+
+This is what we want to achieve with this proposal:
+
+* policy authors: have a way to sign their policies, to attest their provenance.
+* end users: have a way to verify the integrity of a policy that they want to load inside of their cluster. Allow them to execute only policies they trust.
+
+## Examples / User Stories
+[examples]: #examples
+
+> Examples of how the feature will be used. Interactions should show the action and the response. When appropriate, provide user stories in the form of "As a [role], I want [feature], so [that]."
+
+### Execute only policies coming from trusted parties
+
+> As a Kubernates administrator,
+> I want to enforce only Kubewarden policies that have been produced by authors I trust to be loaded.
+
+### Execute only policies that have been approved by selected people
+
+> As a group of Kubernetes administrators,
+> we want to deploy in production only policies that we have approved.
+
+In that case, the Kubernetes administrators would:
+
+  * Download a pre-built Kubewarden policy
+  * Verify it's being signed by its author
+  * Push it to an internal registry
+  * All the Kubernetes administrators would sign the policy
+  * The policy is loaded only if all the Kubernetes administrators signed the policy
+
+
+# Detailed design
+[design]: #detailed-design
+
+> This is the bulk of the RFC. Explain the design in enough detail for somebody familiar with the product to understand, and for somebody familiar with the internals to implement.
+>
+> This section should cover architecture aspects and the rationale behind disruptive technical decisions (when applicable), as well as corner-cases and warnings.
+
+We're going to use [Sigstore](https://sigstore.dev) to sign and verify our policies.
+
+Sigstore allows policies to be signed either with a pub/private keypair or with
+[KEYLESS](https://github.com/sigstore/cosign/blob/main/KEYLESS.md)
+signing.
+
+## How Sigstore signature work
+
+This is a quick overview of the different kind of signatures that could be produced by Sigstore.
+
+Even though we will use just one method to sign our policies, we will have to support
+all of them in order to allow the use cases described above. We don't know how users and 3rd party
+developer will sign their policies.
+
+It's important to know that a policy can be signed in multiple ways. Multiple people can sign the
+policy. The same user can sign it multiple times, using different methods.
+
+### Public/private key pair
+
+This is the simplest way of signing. The policy author uses his private key to sign
+the policy.
+
+The end user needs the public key of the author to verify the policy.
+
+### Keyless signing outside of GitHub Action
+
+The policy author signs the policy in keyless mode. This is the workflow:
+
```

Comment:
This workflow is ok for an initial implementation but needs to be simplified (from an authors pov) in future iterations

Reply:
JFYI: this is a workflow that is imposed by `cosign`. Keyless singning inside of GitHub actions is super transparent, I guess this is what policy authors will use the most.